docs/sources/helm-charts/mimir-distributed/configure/configure-autoscaling.md

```diff
@@ -0,0 +1,194 @@
+
+- An operator and external metrics server
+- Support for multiple metric sources, including Prometheus
+- Custom resources (`ScaledObject`) that define scaling parameters
+- Automatic HPA resource management
+
+For more information, refer to [KEDA documentation](https://keda.sh).
+
+## Configure autoscaling for a new installation
+
+This section describes how to enable autoscaling when deploying Mimir for the first time.
+
+### Steps
+
+1. Configure the Prometheus metrics source in your values file:
+
+   ```yaml
+   kedaAutoscaling:
+     prometheusAddress: "http://prometheus.monitoring:9090"
+     pollingInterval: 10
+   ```
+
+2. Enable and configure autoscaling for desired components:
+
+   ```yaml
+   querier:
+     kedaAutoscaling:
+       enabled: true
+       minReplicaCount: 2
+       maxReplicaCount: 10
+   ```
+
+3. Deploy Mimir using Helm:
+   ```bash
+   helm upgrade --install mimir grafana/mimir-distributed -f values.yaml
+   ```
+
+### Expected outcome
+
+After deployment:
+
+- KEDA creates ScaledObject resources for configured components
+- HPA resources are automatically created and begin monitoring metrics
+- Components scale based on configured thresholds and behaviors
+
+## Migrate existing deployments to autoscaling
+
+This section describes how to safely enable autoscaling for an existing Mimir deployment.
+
+{{< admonition type="warning" >}}
+Migrating to autoscaling carries risks for cluster availability:
+
+1. Autoscaling support in the Helm chart is currently experimental
+2. Enabling autoscaling removes the `replicas` field from Deployments
+3. If KEDA/HPA hasn't started autoscaling the Deployment yet, Kubernetes interprets no replicas as meaning 1 replica
+4. This can cause an outage if the transition is not handled carefully
+5. If you're using GitOps tools like FluxCD or ArgoCD, additional steps might be required to handle the transition smoothly
+
+Consider testing the migration in a non-production environment first.
+{{< /admonition >}}
+
+### Before you begin
+
+- Back up your current Helm values
+- Plan for potential service disruption
+- Consider testing in a non-production environment first
+- Ensure you have a rollback plan ready
+- Consider migrating one component at a time to minimize risk
+
+### Steps
+
+1. Add autoscaling configuration with `preserveReplicas` enabled:
+
+   ```yaml
+   querier:
+     kedaAutoscaling:
+       enabled: true
+       preserveReplicas: true # Maintains stability during migration
+       # ... autoscaling configuration ...
+   ```
+
+2. Apply the changes and verify KEDA setup:
+
+   ```bash
+   # Apply changes
+   helm upgrade mimir grafana/mimir-distributed -f values.yaml
+
+   # Verify setup
+   kubectl get hpa
+   kubectl get scaledobject
+   kubectl describe hpa
+   ```
+
+3. After confirming KEDA is managing scaling (wait at least 2-3 polling intervals), remove `preserveReplicas`:
```

Comment:
```suggestion
3. After confirming that KEDA is managing scaling (wait at least 2-3 polling intervals), remove `preserveReplicas`:
```

Reply:
This seems like it could be divided into two separate steps. 
1. Wait 2-3 polling intervals to confirm that KEDA is managing scaling.
2. Remove `preservedReplicas`.

Reply:
```suggestion
3. Wait 2-3 polling intervals to confirm that KEDA is managing scaling.
4. Remove `preserveReplicas`.
```